test/pkg/environment/azure/environment.go

```diff
@@ -105,11 +110,28 @@ func NewEnvironment(t *testing.T) *Environment {
 	azureEnv.NodeResourceGroup = defaultNodeRG
 
 	cred := lo.Must(azidentity.NewDefaultAzureCredential(nil))
+	azureEnv.defaultCredential = cred
+	byokRetryOptions := azureEnv.ClientOptionsForRBACPropagation()
+	azureEnv.vmClient = lo.Must(armcompute.NewVirtualMachinesClient(azureEnv.SubscriptionID, cred, nil))
+	azureEnv.vnetClient = lo.Must(armnetwork.NewVirtualNetworksClient(azureEnv.SubscriptionID, cred, nil))
+	azureEnv.subnetClient = lo.Must(armnetwork.NewSubnetsClient(azureEnv.SubscriptionID, cred, nil))
+	azureEnv.interfacesClient = lo.Must(armnetwork.NewInterfacesClient(azureEnv.SubscriptionID, cred, nil))
+	azureEnv.managedClusterClient = lo.Must(containerservice.NewManagedClustersClient(azureEnv.SubscriptionID, cred, nil))
+	azureEnv.KeyVaultClient = lo.Must(armkeyvault.NewVaultsClient(azureEnv.SubscriptionID, cred, byokRetryOptions))
+	azureEnv.DiskEncryptionSetClient = lo.Must(armcompute.NewDiskEncryptionSetsClient(azureEnv.SubscriptionID, cred, byokRetryOptions))
+	azureEnv.RBACManager = lo.Must(NewRBACManager(azureEnv.SubscriptionID, cred))
+	return azureEnv
+}
 
-	// Retry options for BYOK-related clients that may encounter RBAC propagation delays
-	// RBAC assignments can take time to propagate, resulting in 403 Forbidden errors
-	// With 15 retries at 5 second intervals = 75 seconds total retry time
-	byokRetryOptions := &arm.ClientOptions{
+func (env *Environment) GetDefaultCredential() azcore.TokenCredential {
```

Comment:
minor, style: I don't think we care that this is from DefaultCredential. It's just the credential. I would just call it `GetCredential`?